docs/sphinx/source/singlediode.rst

```diff
@@ -0,0 +1,104 @@
+.. _singlediode:
+
+Single Diode Model
+==================
+
+This section reviews the solutions to the single diode model used in
+pvlib-python to generate an IV curve of a PV module.
+
+pvlib-python supports two ways to solve the single diode model, by passing the
+a ``method`` keyword to the :func:`pvlib.pvsystem.singlediode` function:
+
+1. Lambert W-Function
+2. Bishop's Algorithm
+
+Lambert W-Function
+------------------
+When ``method='lambertw'``, the Lambert W-function is used as previously shown
+by Jain and Kapoor [1, 2]. The following algorithm can be found on
+`Wikipedia: Theory of Solar Cells
+<https://en.wikipedia.org/wiki/Theory_of_solar_cells>`_, given the basic single
+diode model equation.
+
+.. math::
+
+   I = I_L - I_0 \left(\exp \left(\frac{\left(V + I R_s \right)}{n Ns V_{th}} \right) - 1 \right)
+       - \frac{\left(V + I R_s \right)}{R_{sh}}
+
+Lambert W-function is the inverse of the function
+:math:`f \left( w \right) = w \exp \left( w \right)` or
+:math:`w = f^{-1} \left( w \exp \left( w \right) \right)` also given as
+:math:`w = W \left( w \exp \left( w \right) \right)`. Rearranging the single
+diode equation above with a Lambert W-function yields the following.
```

Comment:
I'd replace the phrase "Rearranging the single diode equation above with a Lambert W-function yields the following. " with "Defining"

Because I could not find a derivation of the Lambert W solution in literature, I wrote it out in a Sandia report. We could add a reference if you want.

Reply:
Is it this reference? ["Parameter Estimation for Single Diode Models of Photovoltaic Modules," Clifford W. Hansen, Sandia Report SAND2015-2065, 2015](https://prod.sandia.gov/techlib-noauth/access-control.cgi/2015/152065.pdf)

Reply:
✅ OK, I think I've reworded it using "Defining ..." is this better:
>Defining the following parameter, :math:`z`, is necessary to transform the single diode equation into a form that can be expressed as a Lambert W-function.